moto/backup/models.py

```diff
@@ -0,0 +1,204 @@
+"""BackupBackend class with methods for supported APIs."""
+
+from typing import Dict, Any, List
+
+from moto.core import BaseBackend, BackendDict, BaseModel
+from moto.utilities.tagging_service import TaggingService
+from moto.core.utils import unix_time
+from moto.moto_api._internal import mock_random
+from .exceptions import (
+    AlreadyExistsException,
+    ResourceNotFoundException,
+)
+
+class Plan(BaseModel):
+    def __init__(
+        self,
+        backup_plan: Dict[str, Any],
+        creator_request_id: str,
+        backend: "BackupBackend",
+    ):
+        self.backup_plan_id = str(mock_random.uuid4())
+        self.backup_plan_arn = f"arn:aws:backup:{backend.region_name}:{backend.account_id}:backup-plan:{self.backup_plan_id}"
+        self.creation_date = unix_time()
+        self.version_id = self.random_str_utf8()
+        self.creator_request_id = creator_request_id
+        self.backup_plan = backup_plan
+        adv_settings = backup_plan.get("AdvancedBackupSettings")
+        self.advanced_backup_settings = adv_settings or []
+
+    def to_dict(self) -> Dict[str, Any]:
+        dct = {
+            "BackupPlanId": self.backup_plan_id,
+            "BackupPlanArn": self.backup_plan_arn,
+            "CreationDate": self.creation_date,
+            "VersionId": self.version_id,
+            "AdvancedBackupSettings": self.advanced_backup_settings,
+        }
+        return {k: v for k, v in dct.items() if v}
+    
+    def to_get_dict(self) -> Dict[str, Any]:
+        dct = self.to_dict()
+        dct_options = {
+            "BackupPlan": self.backup_plan,
+            "CreatorRequestId": self.creator_request_id,
+            "DeletionDate": unix_time(),# need to find deletion and lastexecutiondate
+            "LastExecutionDate": unix_time()
+        }
+        for key, value in dct_options.items():
+            if value is not None:
+                dct[key] = value
+        return dct
+    
+    def to_list_dict(self) -> Dict[str, Any]:
+        dct = self.to_get_dict()
+        dct.pop("BackupPlan")
+        dct["BackupPlanName"] = self.backup_plan.get("BackupPlanName")
+        return dct
+    
+    def random_str_utf8(self) -> str:
+        ran_str = mock_random.get_random_string(length=48)
+        utf8_bytes = ran_str.encode('utf-8')
+        return utf8_bytes.decode('utf-8')
```

Comment:
What is the reasoning behind the decode/encode operation? It looks like we can just `return mock_random.get_random_string(length=48)` to get the same outcome. (Or even get rid of the function altogether..)